xtable-core/src/test/java/org/apache/xtable/paimon/TestPaimonSchemaExtractor.java

```diff
@@ -0,0 +1,496 @@
+    InternalField expectedField =
+        InternalField.builder()
+            .name("double_field")
+            .fieldId(8)
+            .schema(
+                InternalSchema.builder()
+                    .name("DOUBLE")
+                    .dataType(InternalType.DOUBLE)
+                    .isNullable(true)
+                    .build())
+            .defaultValue(InternalField.Constants.NULL_DEFAULT_VALUE)
+            .build();
+    assertField(paimonField, expectedField);
+  }
+
+  @Test
+  void testDateField() {
+    DataField paimonField = new DataField(9, "date_field", new DateType());
+    InternalField expectedField =
+        InternalField.builder()
+            .name("date_field")
+            .fieldId(9)
+            .schema(
+                InternalSchema.builder()
+                    .name("DATE")
+                    .dataType(InternalType.DATE)
+                    .isNullable(true)
+                    .build())
+            .defaultValue(InternalField.Constants.NULL_DEFAULT_VALUE)
+            .build();
+    assertField(paimonField, expectedField);
+  }
+
+  @Test
+  void testTimestampField() {
+    DataField paimonField = new DataField(10, "timestamp_field", new TimestampType(3));
+    Map<InternalSchema.MetadataKey, Object> timestampMetadata =
+        Collections.singletonMap(
+            InternalSchema.MetadataKey.TIMESTAMP_PRECISION, InternalSchema.MetadataValue.MICROS);
+    InternalField expectedField =
+        InternalField.builder()
+            .name("timestamp_field")
+            .fieldId(10)
+            .schema(
+                InternalSchema.builder()
+                    .name("TIMESTAMP(3)")
+                    .dataType(InternalType.TIMESTAMP)
+                    .isNullable(true)
+                    .metadata(timestampMetadata)
+                    .build())
+            .defaultValue(InternalField.Constants.NULL_DEFAULT_VALUE)
+            .build();
+    assertField(paimonField, expectedField);
+  }
+
+  @Test
+  void testDecimalField() {
+    DataField paimonField = new DataField(11, "decimal_field", new DecimalType(10, 2));
+    Map<InternalSchema.MetadataKey, Object> decimalMetadata = new HashMap<>();
+    decimalMetadata.put(InternalSchema.MetadataKey.DECIMAL_PRECISION, 10);
+    decimalMetadata.put(InternalSchema.MetadataKey.DECIMAL_SCALE, 2);
+    InternalField expectedField =
+        InternalField.builder()
+            .name("decimal_field")
+            .fieldId(11)
+            .schema(
+                InternalSchema.builder()
+                    .name("DECIMAL(10, 2)")
+                    .dataType(InternalType.DECIMAL)
+                    .isNullable(true)
+                    .metadata(decimalMetadata)
+                    .build())
+            .defaultValue(InternalField.Constants.NULL_DEFAULT_VALUE)
+            .build();
+    assertField(paimonField, expectedField);
+  }
+
+  @Test
+  void testStructField() {
```

Comment:
Nitpick: for my own sanity, I usually like to go three levels deep to ensure that the parent path is not simply the name of the parent field but rather the full path to that field. Can we do that here?